Unable to compile libzim in fresh Ubuntu 22.04

```
The Meson build system
Version: 0.61.2
Source dir: /home/kelson/code/libzim
Build dir: /home/kelson/code/libzim/build
Build type: native build
Project name: libzim
Project version: 8.2.1
C compiler for the host machine: cc (gcc 11.4.0 "cc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0")
C linker for the host machine: cc ld.bfd 2.38
C++ compiler for the host machine: c++ (gcc 11.4.0 "c++ (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0")
C++ linker for the host machine: c++ ld.bfd 2.38
Host machine cpu family: x86_64
Host machine cpu: x86_64
Checking for size of "off_t" : 8
Checking for size of "size_t" : 8
Found pkg-config: /usr/bin/pkg-config (0.29.2)
Run-time dependency liblzma found: YES 5.2.5
Run-time dependency libzstd found: YES 1.4.8
Run-time dependency xapian-core found: YES 1.4.18
Run-time dependency threads found: YES
Run-time dependency icu-i18n found: YES 70.1
Run-time dependency GTest found: YES (building self)
Configuring zim_config.h using configuration
Program libzim-compile-resources found: YES (/home/kelson/code/libzim/scripts/libzim-compile-resources)
Program download_test_data.py found: YES (/home/kelson/code/libzim/scripts/download_test_data.py)
Configuring config.h using configuration
Build targets in project: 33

Found ninja-1.10.1 at /usr/bin/ninja
ninja: Entering directory `build'
[1/191] Generating static/resources with a custom command
stopwords/af ['stopwords', 'af']
stopwords/ar ['stopwords', 'ar']
stopwords/bg ['stopwords', 'bg']
stopwords/bn ['stopwords', 'bn']
stopwords/br ['stopwords', 'br']
stopwords/ca ['stopwords', 'ca']
stopwords/cs ['stopwords', 'cs']
stopwords/da ['stopwords', 'da']
stopwords/de ['stopwords', 'de']
stopwords/el ['stopwords', 'el']
stopwords/en ['stopwords', 'en']
stopwords/eo ['stopwords', 'eo']
stopwords/es ['stopwords', 'es']
stopwords/et ['stopwords', 'et']
stopwords/eu ['stopwords', 'eu']
stopwords/fa ['stopwords', 'fa']
stopwords/fi ['stopwords', 'fi']
stopwords/fr ['stopwords', 'fr']
stopwords/ga ['stopwords', 'ga']
stopwords/gl ['stopwords', 'gl']
stopwords/gu ['stopwords', 'gu']
stopwords/ha ['stopwords', 'ha']
stopwords/he ['stopwords', 'he']
stopwords/hi ['stopwords', 'hi']
stopwords/hr ['stopwords', 'hr']
stopwords/hu ['stopwords', 'hu']
stopwords/hy ['stopwords', 'hy']
stopwords/id ['stopwords', 'id']
stopwords/it ['stopwords', 'it']
stopwords/ja ['stopwords', 'ja']
stopwords/ko ['stopwords', 'ko']
stopwords/ku ['stopwords', 'ku']
stopwords/la ['stopwords', 'la']
stopwords/lt ['stopwords', 'lt']
stopwords/lv ['stopwords', 'lv']
stopwords/mr ['stopwords', 'mr']
stopwords/ms ['stopwords', 'ms']
stopwords/nl ['stopwords', 'nl']
stopwords/no ['stopwords', 'no']
stopwords/pl ['stopwords', 'pl']
stopwords/pt ['stopwords', 'pt']
stopwords/ro ['stopwords', 'ro']
stopwords/ru ['stopwords', 'ru']
stopwords/sk ['stopwords', 'sk']
stopwords/sl ['stopwords', 'sl']
stopwords/so ['stopwords', 'so']
stopwords/st ['stopwords', 'st']
stopwords/sv ['stopwords', 'sv']
stopwords/sw ['stopwords', 'sw']
stopwords/th ['stopwords', 'th']
stopwords/tl ['stopwords', 'tl']
stopwords/tr ['stopwords', 'tr']
stopwords/uk ['stopwords', 'uk']
stopwords/ur ['stopwords', 'ur']
stopwords/vi ['stopwords', 'vi']
stopwords/yo ['stopwords', 'yo']
stopwords/zh ['stopwords', 'zh']
stopwords/zu ['stopwords', 'zu']
[64/191] Compiling C++ object test/error_in_creator.p/error_in_creator.cpp.o
FAILED: test/error_in_creator.p/error_in_creator.cpp.o 
c++ -Itest/error_in_creator.p -Iinclude -I../include -Isrc -I../src -I/usr/src/googletest/googletest -I/usr/src/googletest/googletest/include -fdiagnostics-color=always -Wall -Winvalid-pch -Wnon-virtual-dtor -Werror -std=c++17 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -pthread -DWITH_TEST_DATA=1 -MD -MQ test/error_in_creator.p/error_in_creator.cpp.o -MF test/error_in_creator.p/error_in_creator.cpp.o.d -o test/error_in_creator.p/error_in_creator.cpp.o -c ../test/error_in_creator.cpp
In file included from /usr/src/googletest/googletest/include/gtest/gtest.h:67,
                 from ../test/error_in_creator.cpp:28:
/usr/src/googletest/googletest/include/gtest/gtest-param-test.h:500:38: error: ‘constexpr bool testing::internal::InstantiateTestCase_P_IsDeprecated()’ is deprecated: INSTANTIATE_TEST_CASE_P is deprecated, please use INSTANTIATE_TEST_SUITE_P [-Werror=deprecated-declarations]
  500 |   static_assert(::testing::internal::InstantiateTestCase_P_IsDeprecated(), \
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/src/googletest/googletest/include/gtest/gtest-param-test.h:500:38: note: in definition of macro ‘INSTANTIATE_TEST_CASE_P’
  500 |   static_assert(::testing::internal::InstantiateTestCase_P_IsDeprecated(), \
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/src/googletest/googletest/include/gtest/gtest.h:62,
                 from ../test/error_in_creator.cpp:28:
/usr/src/googletest/googletest/include/gtest/internal/gtest-internal.h:1297:16: note: declared here
 1297 | constexpr bool InstantiateTestCase_P_IsDeprecated() { return true; }
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/src/googletest/googletest/include/gtest/gtest.h:67,
                 from ../test/error_in_creator.cpp:28:
/usr/src/googletest/googletest/include/gtest/gtest-param-test.h:500:72: error: ‘constexpr bool testing::internal::InstantiateTestCase_P_IsDeprecated()’ is deprecated: INSTANTIATE_TEST_CASE_P is deprecated, please use INSTANTIATE_TEST_SUITE_P [-Werror=deprecated-declarations]
  500 |   static_assert(::testing::internal::InstantiateTestCase_P_IsDeprecated(), \
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
../test/error_in_creator.cpp:190:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
  190 | INSTANTIATE_TEST_CASE_P(
      | ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/src/googletest/googletest/include/gtest/gtest.h:62,
                 from ../test/error_in_creator.cpp:28:
/usr/src/googletest/googletest/include/gtest/internal/gtest-internal.h:1297:16: note: declared here
 1297 | constexpr bool InstantiateTestCase_P_IsDeprecated() { return true; }
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/src/googletest/googletest/include/gtest/gtest.h:67,
                 from ../test/error_in_creator.cpp:28:
/usr/src/googletest/googletest/include/gtest/gtest-param-test.h:500:38: error: ‘constexpr bool testing::internal::InstantiateTestCase_P_IsDeprecated()’ is deprecated: INSTANTIATE_TEST_CASE_P is deprecated, please use INSTANTIATE_TEST_SUITE_P [-Werror=deprecated-declarations]
  500 |   static_assert(::testing::internal::InstantiateTestCase_P_IsDeprecated(), \
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/src/googletest/googletest/include/gtest/gtest-param-test.h:500:38: note: in definition of macro ‘INSTANTIATE_TEST_CASE_P’
  500 |   static_assert(::testing::internal::InstantiateTestCase_P_IsDeprecated(), \
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/src/googletest/googletest/include/gtest/gtest.h:62,
                 from ../test/error_in_creator.cpp:28:
/usr/src/googletest/googletest/include/gtest/internal/gtest-internal.h:1297:16: note: declared here
 1297 | constexpr bool InstantiateTestCase_P_IsDeprecated() { return true; }
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/src/googletest/googletest/include/gtest/gtest.h:67,
                 from ../test/error_in_creator.cpp:28:
/usr/src/googletest/googletest/include/gtest/gtest-param-test.h:500:72: error: ‘constexpr bool testing::internal::InstantiateTestCase_P_IsDeprecated()’ is deprecated: INSTANTIATE_TEST_CASE_P is deprecated, please use INSTANTIATE_TEST_SUITE_P [-Werror=deprecated-declarations]
  500 |   static_assert(::testing::internal::InstantiateTestCase_P_IsDeprecated(), \
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
../test/error_in_creator.cpp:360:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
  360 | INSTANTIATE_TEST_CASE_P(
      | ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/src/googletest/googletest/include/gtest/gtest.h:62,
                 from ../test/error_in_creator.cpp:28:
/usr/src/googletest/googletest/include/gtest/internal/gtest-internal.h:1297:16: note: declared here
 1297 | constexpr bool InstantiateTestCase_P_IsDeprecated() { return true; }
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
[73/191] Compiling C++ object test/header.p/_usr_src_googletest_googletest_src_gtest-all.cc.o
ninja: build stopped: subcommand failed.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -170,7 +170,7 @@ jobs:
       HOME: /home/runner
     runs-on: ubuntu-22.04
     container:
-      image: "ghcr.io/kiwix/kiwix-build_ci_${{matrix.image_variant}}:38"
+      image: "ghcr.io/kiwix/kiwix-build_ci_${{matrix.image_variant}}:2023-10-30"
 
     steps:
       - name: Install dependences

--- a/.github/workflows/package.yml
+++ b/.github/workflows/package.yml
@@ -17,7 +17,6 @@ jobs:
         distro:
           - debian-unstable
           - debian-bullseye
-          - debian-buster
           - ubuntu-jammy
           - ubuntu-focal
     steps:
@@ -53,13 +52,6 @@ jobs:
         if: matrix.distro == 'debian-bullseye'
         name: Build package for debian-bullseye
         id: build-debian-bullseye
-        with:
-          args: --no-sign
-
-      - uses: legoktm/gh-action-build-deb@debian-buster
-        if: matrix.distro == 'debian-buster'
-        name: Build package for debian-buster
-        id: build-debian-buster
         with:
           args: --no-sign
 

--- a/meson.build
+++ b/meson.build
@@ -80,7 +80,7 @@ else
     icu_dep = dependency('icu-i18n', required:false, static:static_linkage)
 endif
 
-gtest_dep = dependency('gtest', main:true, fallback:['gtest', 'gtest_main_dep'], required:false)
+gtest_dep = dependency('gtest', version: '>=1.10.0', main:true, fallback:['gtest', 'gtest_main_dep'], required:false)
 
 inc = include_directories('include')
 

--- a/subprojects/gtest.wrap
+++ b/subprojects/gtest.wrap
@@ -1,10 +1,16 @@
 [wrap-file]
-directory = googletest-release-1.8.1
+directory = googletest-1.14.0
+source_url = https://github.com/google/googletest/archive/refs/tags/v1.14.0.tar.gz
+source_filename = gtest-1.14.0.tar.gz
+source_hash = 8ad598c73ad796e0d8280b082cebd82a630d73e73cd3c70057938a6501bba5d7
+patch_filename = gtest_1.14.0-1_patch.zip
+patch_url = https://wrapdb.mesonbuild.com/v2/gtest_1.14.0-1/get_patch
+patch_hash = 2e693c7d3f9370a7aa6dac802bada0874d3198ad4cfdf75647b818f691182b50
+source_fallback_url = https://github.com/mesonbuild/wrapdb/releases/download/gtest_1.14.0-1/gtest-1.14.0.tar.gz
+wrapdb_version = 1.14.0-1
 
-source_url = https://github.com/google/googletest/archive/release-1.8.1.zip
-source_filename = gtest-1.8.1.zip
-source_hash = 927827c183d01734cc5cfef85e0ff3f5a92ffe6188e0d18e909c5efebf28a0c7
-
-patch_url = https://wrapdb.mesonbuild.com/v1/projects/gtest/1.8.1/1/get_zip
-patch_filename = gtest-1.8.1-1-wrap.zip
-patch_hash = f79f5fd46e09507b3f2e09a51ea6eb20020effe543335f5aee59f30cc8d15805
+[provide]
+gtest = gtest_dep
+gtest_main = gtest_main_dep
+gmock = gmock_dep
+gmock_main = gmock_main_dep

--- a/test/compression.cpp
+++ b/test/compression.cpp
@@ -39,7 +39,7 @@ using CompressionAlgo = ::testing::Types<
   ZSTD_INFO
 >;
 
-TYPED_TEST_CASE(CompressionTest, CompressionAlgo);
+TYPED_TEST_SUITE(CompressionTest, CompressionAlgo);
 
 TYPED_TEST(CompressionTest, compress) {
   std::string data;

--- a/test/decoderstreamreader.cpp
+++ b/test/decoderstreamreader.cpp
@@ -60,7 +60,7 @@ using CompressionTypes = ::testing::Types<
   ZSTD_INFO
 >;
 
-TYPED_TEST_CASE(DecoderStreamReaderTest, CompressionTypes);
+TYPED_TEST_SUITE(DecoderStreamReaderTest, CompressionTypes);
 
 TYPED_TEST(DecoderStreamReaderTest, justCompressedData) {
   typedef typename TestFixture::CompressionInfo CompressionInfo;

--- a/test/error_in_creator.cpp
+++ b/test/error_in_creator.cpp
@@ -184,10 +184,10 @@ TEST_P(FaultyItemErrorTest, faultyItem)
 
 // It would be more natural to put the `#if defined` only around the
 // discarded values, but when crosscompiling on Windows, compiler fail to
-// understand ``#if defined` when used inside the `INSTANTIATE_TEST_CASE_P`
+// understand ``#if defined` when used inside the `INSTANTIATE_TEST_SUITE_P`
 // macro. I suspect some macro definition conflicts.
 #if defined(ENABLE_XAPIAN)
-INSTANTIATE_TEST_CASE_P(
+INSTANTIATE_TEST_SUITE_P(
 CreatorError,
 FaultyItemErrorTest,
 ::testing::Values(
@@ -200,7 +200,7 @@ FaultyItemErrorTest,
     ERRORKIND::GET_INDEXDATA
 ));
 #else
-INSTANTIATE_TEST_CASE_P(
+INSTANTIATE_TEST_SUITE_P(
 CreatorError,
 FaultyItemErrorTest,
 ::testing::Values(
@@ -354,10 +354,10 @@ TEST_P(FaultyDelayedItemErrorTest, faultyUnfinishedCreator)
 }
 // It would be more natural to put the `#if defined` only around the
 // discarded values, but when crosscompiling on Windows, compiler fail to
-// understand ``#if defined` when used inside the `INSTANTIATE_TEST_CASE_P`
+// understand ``#if defined` when used inside the `INSTANTIATE_TEST_SUITE_P`
 // macro. I suspect some macro definition conflicts.
 #if defined(ENABLE_XAPIAN)
-INSTANTIATE_TEST_CASE_P(
+INSTANTIATE_TEST_SUITE_P(
 CreatorError,
 FaultyDelayedItemErrorTest,
 ::testing::Values(
@@ -372,7 +372,7 @@ FaultyDelayedItemErrorTest,
     ERRORKIND::GET_INDEXDATA_POSITION
 ));
 #else
-INSTANTIATE_TEST_CASE_P(
+INSTANTIATE_TEST_SUITE_P(
 CreatorError,
 FaultyDelayedItemErrorTest,
 ::testing::Values(

